### PR TITLE
Build the rest of the signed-out home page components

### PR DESCRIPTION
**Pull request by goplayoutside3**
## Package
lib-content

## Linked Issue and/or Talk Post
Toward: https://github.com/zooniverse/front-end-monorepo/issues/6024
Figma: https://www.figma.com/design/p6laHCEklj9YK79TC1F7NV/Homepages

## Describe your changes
- Build FeaturedProjects component
- Build Researchers component
- Add container styling to Community for a seamless page layout

## How to Review

There are still a few styling adjustments I need to make in order to exactly match the Figma design linked above, but I'll comment about those below.

To review, run app-root locally and visit https://local.zooniverse.org:3000 while signed-out. Try out light and dark mode, and various viewport widths.

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General
- [ ] Tests are passing locally and on Github
- [ ] Documentation is up to date and changelog has been updated if appropriate
- [ ] You can `yarn panic && yarn bootstrap` or `docker-compose up --build` and FEM works as expected
- [ ] FEM works in all major desktop browsers: Firefox, Chrome, Edge, Safari (Use Browserstack account as needed)
- [ ] FEM works in a mobile browser

### CODE REVIEW

File: packages/lib-content/src/screens/Home/DefaultHome/DefaultHome.js [hunk 1]
```diff
@@ -21,11 +29,24 @@ export default function DefaultHome() {
...
+        <SpacedHeading
+          level={2}
+          size='1.5rem'
+          color={{ light: 'neutral-1', dark: 'accent-1' }}
+          textAlign='center'
+          fill
+        >
+          {t('Home.DefaultHome.headings.four')}
+        </SpacedHeading>
+        <SubHeading>{t('Home.DefaultHome.subheadings.four')}</SubHeading>
```
Comment (goplayoutside3) [inline packages/lib-content/src/screens/Home/DefaultHome/DefaultHome.js:48]: SubHeading font size needs a bit of adjustment to adapt to smaller viewport widths. Will do in a separate PR because SubHeading is used throughout the About Zooniverse pages too.

File: packages/lib-content/src/screens/Home/DefaultHome/components/FeaturedProjects.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  return useSWR('featured-projects', getFeaturedProjects, SWROptions)
+}
+
+export default function FeaturedProjects() {
+  const { data: featuredProjects, isLoading } = useFeaturedProjects()
+  const { t } = useTranslation()
+
+  // A breakpoint x-large needs to be implemented at the theme level
+  // Then on x-large can be passed to ProjectCard below
+  const size = useContext(ResponsiveContext)
```
Comment (goplayoutside3) [inline packages/lib-content/src/screens/Home/DefaultHome/components/FeaturedProjects.js:50]: I want to be able to simply past `size` to ProjectCard, which is a prop that determines its width. We don't have an `x-large` breakpoint in FEM's Grommet theme yet, so that will need to be added in a separate PR. For now, the featured projects section has cards sized large, medium, and small.

Comment (goplayoutside3) [reply packages/lib-content/src/screens/Home/DefaultHome/components/FeaturedProjects.js]: I actually decided to just write the ProjectCard size logic directly in FeaturedProjects rather than rely on an 'xlarge' Grommet theme breakpoint. See https://github.com/zooniverse/front-end-monorepo/pull/6109 for our current breakpoint sizes.

File: packages/lib-content/src/screens/Home/DefaultHome/components/Researchers.js [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+  aspect-ratio: 16 / 9;
+  overflow: hidden;
+
+  @media (width <= 769px) {
+    min-width: 370px;
+  }
+`
+
+const YOUTUBE_SOURCES = [
+  'https://www.youtube-nocookie.com/embed/CaTNIoJy4Dg?si=-rG8luvWD_FtFBc5', // PH TESS
```
Comment (shaunanoordin) [inline packages/lib-content/src/screens/Home/DefaultHome/components/Researchers.js:65]: [rambling] This is my first time seeing `www.youtube-nocookie.com`. I tried searching to find more info on it, and found the official documentation _buried very low_ on the Google search results under YouTube Help -> Embed videos and playlists -> Turn on privacy-enhanced mode. _Fascinating._

<details>
<summary>For posterity, here's what's written on that YouTube "Embed videos and playlists" help page's "Turn on privacy-enhanced mode" section, as of 6 Jun 2024</summary>

The privacy-enhanced mode of the YouTube embedded player prevents the use of views of embedded YouTube content from influencing the viewer's browsing experience on YouTube. This means that the view of a video shown in the privacy-enhanced mode of the embedded player will not be used to personalise the YouTube browsing experience, either within your privacy-enhanced mode embedded player or in the viewer's subsequent YouTube viewing experience.

If ads are served on a video shown in the privacy-enhanced mode of the embedded player, those ads will likewise be non-personalised. In addition, the view of a video shown in the privacy-enhanced mode of the embedded player will not be used to personalise advertising shown to the viewer outside of your site or app.

As a reminder, the [YouTube API Terms of Service](https://developers.google.com/youtube/terms/api-services-terms-of-service) and [developer policies](https://developers.google.com/youtube/terms/developer-policies) apply to the access and use of the YouTube embedded player.

To use privacy-enhanced mode:

Change the domain for the embed URL in your HTML from https://www.youtube.com to https://www.youtube-nocookie.com.
For network administrators: You'll need to add youtube-nocookie.com to the firewall allowlist.
For use in applications, use a WebView instance of the embedded player. Privacy-enhanced mode is currently available only for embedded players on websites.
Important: If your website or app is child-directed, you must self-designate your site or app using [these tools](https://support.google.com/policies/answer/9664901), as required by the [YouTube API Terms of Service](https://developers.google.com/youtube/terms/api-services-terms-of-service) and [developer policies](https://developers.google.com/youtube/terms/developer-policies#j.-child-directed-api-clients), even if you embed YouTube videos with the privacy-enhanced mode player.
Example:

Before

<iframe width="1440" height="762"

src="https://www.youtube.com/embed/7cjVj1ZyzyE"

frameborder="0" allow="autoplay; encrypted-media" allowfullscreen></iframe>

After

<iframe width="1440" height="762" src="https://www.youtube-nocookie.com/embed/7cjVj1ZyzyE"

frameborder="0" allow="autoplay; encrypted-media" allowfullscreen></iframe>
</details>

File: packages/lib-content/src/screens/Home/DefaultHome/components/Researchers.js [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+  'https://www.youtube-nocookie.com/embed/8lZiZoBcMjE?si=wPKWlRjQcKeTIOB5', // Wildcam Gorongosa
+  'https://www.youtube.com/embed/B6diEbpEblQ?si=tt3jtcINdleiBT8g' // Backyard Worlds
```
Comment (shaunanoordin) [inline packages/lib-content/src/screens/Home/DefaultHome/components/Researchers.js:67]: [minor] Any reason this one has no no-cookie? Making it a yes-cookie by boolean logic, I guess.

File: packages/lib-content/src/screens/Home/DefaultHome/components/Researchers.js [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+        >
+          {YOUTUBE_SOURCES.map(src => (
+            <VideoWrapper>
+              <iframe
+                width='100%'
+                height='100%'
+                src={src}
+                title='TESS YouTube video player'
+                frameborder='0'
+                allowfullscreen
```
Comment (shaunanoordin) [inline packages/lib-content/src/screens/Home/DefaultHome/components/Researchers.js:104]: [minor] React's getting fussy about these variables in the dev console, insisting it should be called `frameBorder` and `allowFullScreen`. 🤷 (Which is mildly confusing, because 1. lower caps is the correct way to write it in HTML, 2. `frameborder="0"` is being rendered in the DOM anyway and the iframe's frames are indeed not showing, 3. but somehow allowfullscreen isn't being rendered in the DOM so the YT videos can't go full screen.)

File: packages/lib-content/src/screens/Home/DefaultHome/components/Researchers.js [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+        <Box
+          direction='row'
+          justify='between'
+          margin={{ top: 'medium' }}
+          gap='small'
+          pad={{ horizontal: 'xxsmall', bottom: 'xsmall' }}
+          overflow={{ horizontal: 'auto' }}
+        >
+          {YOUTUBE_SOURCES.map(src => (
+            <VideoWrapper>
```
Comment (shaunanoordin) [inline packages/lib-content/src/screens/Home/DefaultHome/components/Researchers.js:97]: I think this is missing a `key`

File: packages/lib-content/src/screens/Home/DefaultHome/components/FeaturedProjects.js [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+              <ProjectCard
+                key={project.slug}
+                description={project.description}
+                displayName={project.display_name}
+                href={project.slug}
+                imageSrc={project.avatar_src}
+                size={cardSize}
+              />
+            ))
+          : null}
```
Comment (shaunanoordin) [inline packages/lib-content/src/screens/Home/DefaultHome/components/FeaturedProjects.js:86]: [minor][non-blocking][future dev] I think we should at least have some sort of message to indicate "sorry, no featured projects available!" or something, on the unlikely occasion that we have no featured projects. But that can probably wait for a design pass for future improvement.

File: packages/lib-content/src/screens/Home/DefaultHome/components/FeaturedProjects.js [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+        as='ul'
+        direction='row'
+        justify='between'
+        margin={{ top: 'medium', bottom: '0' }}
+        gap='small'
+        pad={{ horizontal: 'xxsmall', bottom: 'xsmall' }}
+        overflow={{ horizontal: 'auto' }}
+        style={{ listStyle: 'none' }}
+      >
+        {featuredProjects?.length
```
Comment (shaunanoordin) [inline packages/lib-content/src/screens/Home/DefaultHome/components/FeaturedProjects.js:75]: [minor][rambling] I should figure out where the _"there are exactly **4** (i.e. four (as in 2x2 (the integer between 3 and 5))) Featured Projects at any time, no more, no less"_ rule is being implemented.

As it stands, if panoptes_js returns 100 featured projects for some reason, this page will just shrug, go "sure thing boss" and overload the page with projects.

File: packages/lib-content/src/screens/Home/DefaultHome/components/FeaturedProjects.js [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+          ? featuredProjects.map(project => (
+              <ProjectCard
+                key={project.slug}
+                description={project.description}
+                displayName={project.display_name}
+                href={project.slug}
```
Comment (shaunanoordin) [inline packages/lib-content/src/screens/Home/DefaultHome/components/FeaturedProjects.js:81]: ❗ **Bug:** project cards aren't pointing to the correct project URL. For example, on localhost, I'm getting links to `https://local.zooniverse.org:3000/birgus2/western-shield-camera-watch` instead of `https://local.zooniverse.org:3000/projects/birgus2/western-shield-camera-watch`. I think we're missing a `"/projects"` in this href.

**Review by shaunanoordin — APPROVED** [review]
## PR Review

package: `lib-content`

This PR adds additional components to the Signed-Out Home Page. I believe the additions in this PR now makes this page feature-complete(?)

- On the signed-out home page:
  - (extant) Hero: splashy intro for the Zooniverse
  - (new feature) Featured Projects: fetches featured projects (via panoptes-js) and displays them
      image
    - Clicking on the project card should take users to the appropriate project page.
    - ❗ Bug: the project URL is missing a `"/projects/"`
  - (new feature) Researchers: displays hardcoded selection of embedded videos, and has a link to the lab/project builder.
      image
  - (added existing feature) Mobile: info section advertising our mobile app, with links to the Google & Apple store pages.
  - (extant) Community: lists our recent blog posts

Everything's looking mostly good, save for that one bug!

### Testing

Tested lib-content on localhost with macOS + Chrome 125

Testing methods:
- Open the home page: https://local.zooniverse.org:3000/ (staging)
- Ensure you're signed out. If you're signed in, click the "sign out" option in the top menu.
- Observe all features/components as listed above.
- Modify browser window dimension to ensure website layout is still legible (i.e. check that it looks good on desktop and mobile views)
- Repeat for production version: https://local.zooniverse.org:3000/?env=production (only  featured projects should be different)

### Status

LGTM! I'm approving this PR, though I'd like to see that one small bug fixed before merging.

Bugs to fix:
- [ ] FeaturedProject's ProjectCards should link to the correct project URL.

File: packages/lib-content/src/screens/Home/DefaultHome/components/Researchers.js (hunk 3 above)
Comment (goplayoutside3) [reply packages/lib-content/src/screens/Home/DefaultHome/components/Researchers.js:65]: This is good to know! I think when I grab video URLs to embed on the new homepage, I'm always using Brave so it's likely that browser auto-adds `no-cookie` 🤷‍♀️

File: packages/lib-content/src/screens/Home/DefaultHome/components/Researchers.js (hunk 5 above)
Comment (goplayoutside3) [reply packages/lib-content/src/screens/Home/DefaultHome/components/Researchers.js:104]: Good point! Mark commented on this in https://github.com/zooniverse/front-end-monorepo/pull/6101 too, so I'll correct it for all youtube videos here.

File: packages/lib-content/src/screens/Home/DefaultHome/components/Researchers.js (hunk 6 above)
Comment (goplayoutside3) [reply packages/lib-content/src/screens/Home/DefaultHome/components/Researchers.js:97]: Yes good catch!

File: packages/lib-content/src/screens/Home/DefaultHome/components/FeaturedProjects.js (hunk 8 above)
Comment (goplayoutside3) [reply packages/lib-content/src/screens/Home/DefaultHome/components/FeaturedProjects.js:75]: Agreed. At the moment I think the communications team (either Alisa, or sometimes Sean) sets the desired featured projects via each project's admin page. They manually determine how many projects have the 'featured' flag on.

File: packages/lib-content/src/screens/Home/DefaultHome/components/FeaturedProjects.js (hunk 9 above)
Comment (goplayoutside3) [reply packages/lib-content/src/screens/Home/DefaultHome/components/FeaturedProjects.js:81]: Oh shoot, I'll construct a hardcoded href here instead of just passing the `slug`. Thanks!

File: packages/lib-content/src/screens/Home/DefaultHome/components/FeaturedProjects.js (hunk 7 above)
Comment (goplayoutside3) [reply packages/lib-content/src/screens/Home/DefaultHome/components/FeaturedProjects.js:86]: Good point! I'll add a short message for now.
